client/src/AppBundle/Controller/Ndr/NdrController.php

```diff
@@ -86,18 +132,21 @@ public function indexAction(Redirector $redirector)
     /**
      * @Route("/ndr/{ndrId}/overview", name="ndr_overview")
      * @Template("AppBundle:Ndr/Ndr:overview.html.twig")
+     * @param Redirector $redirector
+     * @return array|RedirectResponse
      */
-    public function overviewAction($ndrId, Redirector $redirector)
+    public function overviewAction(Redirector $redirector)
     {
         // redirect if user has missing details or is on wrong page
-        $user = $this->getUserWithData();
+        $user = $this->userApi->getUserWithData();
         $route = $redirector->getCorrectRouteIfDifferent($user, 'ndr_overview');
 
         if (is_string($route)) {
             return $this->redirectToRoute($route);
         }
 
-        $client = $this->getFirstClient(self::$ndrGroupsForValidation);
+        $user = $this->userApi->getUserWithData(self::$ndrGroupsForValidation);
+        $client = $this->clientApi->getFirstClient($user);
```

Comment:
Believe so. Before we had a function that looked like this

```
protected function getFirstClient($groups = ['foo', 'bar', 'me']) {
    $user = $this->getUserWithData($groups);
    $clients = $user->getClients();

    return (is_array($clients) && !empty($clients[0]) && $clients[0] instanceof Client) ? $clients[0] : null;
}
```

Now we've refactored it to be like
```
public function getFirstClient(User $user) {
    $clients = $user->getClients();

    return (is_array($clients) && !empty($clients[0]) && $clients[0] instanceof Client) ? $clients[0] : null;
}
```

So now instead of passing groups into the function signature we are passing a `User` type so before we call `getFirstClients` now from the API we have to call `getUserWithData` which returns us the user and then we pass that user into `getFirstClient`